packages/mds-policy-author/api.ts

```diff
@@ -188,6 +189,10 @@ function api(app: express.Express): express.Express {
 
   app.post(pathsFor('/policies'), async (req, res) => {
     const policy = req.body
+    if (!isUUID(policy.policy_id)) {
```

Comment:
So, if the body contains an invalid (non-UUID) `policy_id` value then this will overwrite it. I think supplying an invalid API should be an error.

Reply:
Ok, so do you think that the ID generation for a new policy should be the client's responsibility? The reason I cut this PR was because it was unclear whether the `policy_id` would be mandatory from the client or not. I'm pretty indifferent, with a slight preference for the API or DB layer supplying a valid ID if none was given (like normally it's not the client's job to supply a unique ID, when I order something on Amazon I don't pick an ID for that order), but perhaps @avatarneil or @karcass has a stronger opinion.

Reply:
I don't think the client should _have to_ supply a unique ID, but if they _choose to_ supply one then it ought to be valid or the API should return an error.

Reply:
Ah, okay, that finally clicked.